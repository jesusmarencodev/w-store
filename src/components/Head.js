import React from 'react';

import { useAmount } from '../hooks/useAmount';
import { useShoppingCar } from '../hooks/useShoppingCar';
import { useSelector } from 'react-redux';

const Head = () => {
  const { amount } = useAmount();
  const { shoppingCar } = useSelector((state) => state.products);
  const { handlerShowCar } = useShoppingCar();

  return (
    <>
      <header className="head d-flex a-center">
        <div className="head__logo">
          <img className="head__img" src="store.png" alt="store" />
        </div>
        <div
          className={`head__shop d-flex ${shoppingCar && 'head__shop--active'}`}
        >
          <div
            onClick={() => handlerShowCar(true)}
            className={`${
              shoppingCar ? 'shop__button--active' : 'shop__button'
            }`}
          >
            <img className="head__img-car" src="car.png" alt="store" />
            <span className={`shop-total ${shoppingCar && 'shop-active'}`}>
              ${amount}
            </span>
          </div>
          {shoppingCar && (
            <button
              onClick={() => handlerShowCar(false)}
              className="head__shop-button"
            >
              X
            </button>
          )}
        </div>
      </header>
    </>
  );
};

export default Head;
