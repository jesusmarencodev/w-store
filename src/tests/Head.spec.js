import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Head from '../components/Head';

const mockStore = configureStore([]);

describe('Head component', () => {
  it('renders correctly', () => {
    const initialState = {
      products: {
        shoppingCar: false,
      },
    };

    const store = mockStore(initialState);

    const component = renderer.create(
      <Provider store={store}>
        <Head />
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
