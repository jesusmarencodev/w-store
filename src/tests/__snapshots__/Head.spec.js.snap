// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head component renders correctly 1`] = `
<header
  className="head d-flex a-center"
>
  <div
    className="head__logo"
  >
    <img
      alt="store"
      className="head__img"
      src="store.png"
    />
  </div>
  <div
    className="head__shop d-flex false"
  >
    <div
      className="shop__button"
      onClick={[Function]}
    >
      <img
        alt="store"
        className="head__img-car"
        src="car.png"
      />
      <span
        className="shop-total false"
      >
        $
        0
      </span>
    </div>
  </div>
</header>
`;
